src/python/pants/rules/core/test.py

```diff
@@ -29,28 +33,34 @@ class Status(Enum):
     SUCCESS = "SUCCESS"
     FAILURE = "FAILURE"
 
+class CoverageData(ABC):
```

Comment:
Consider adding docstring to explain that subclasses are able to add whatever fields are relevant, such as adding a field for a `snapshot` of the results.